packages/hs-react-ui/src/utils/styles.ts

```diff
@@ -0,0 +1,30 @@
+// A constant factor that works well with base 10 logarithms
+const elevationFactor = 10 ** 0.1;
+
+/**
+ * Calculates the values for shadows based on the provided elevation
+ * @param {number} elevation - The elevation to get values for
+ */
+export const getElevationValues = (elevation = 0) => {
+  const elevationValues = {
+    xOffset: 0,
+    yOffset: 0,
+    blur: 0,
+    opacity: 0,
+  };
+
+  // return the default values of 0 if elevation is 0
+  if (!elevation) {
+    return elevationValues;
+  }
+
+  // Using abs will return the same values for negative and positive elevations
+  const logVal = Math.log10(Math.abs(elevation) + elevationFactor);
+
+  // Set the updated values for the elevation
+  elevationValues.xOffset = 0;
+  elevationValues.yOffset = (logVal / 4) * 0.5;
+  elevationValues.blur = logVal * 0.5;
+  elevationValues.opacity = 0.5 - logVal * 0.2;
+  return elevationValues;
```

Comment:
I'm in two minds about this one - so far all the shadows you updated used the same string with these variables.. should we just return the full shadow string?

The advantage of that approach would mean that we could switch to a box shadow when it's negative and assume that if they want an inset shadow that they're okay with a box-shadow?

Reply:
I was thinking the same thing honestly. I almost did that, but I wasn't sure that made complete sense, but now that you've also mentioned it as well I feel a lot better about returning a string and will update it accordingly (including box-shadow for negative values!)